### PR TITLE
imxrt106x: create script with basic commands and attach it to .data section in plo image

### PR DESCRIPTION
**Pull request by HBuczynski**

### CODE REVIEW

File: _targets/build.project.armv7m7-imxrt106x [hunk 1]
```diff
@@ -32,9 +33,44 @@ export PORTS_JANSSON=y
...
+
+	for prog in "${PROGS[@]}"; do
+		psz=$((($(stat -c "%s" ${PREFIX_PROG_STRIPPED}$prog) + $SIZE_PAGE - 1)&$PAGE_MASK))
+		echo @$prog\($(printf "%x" $poffs):$(printf "%x" $psz)\) >> $PREFIX_BUILD/plo/script.plo
+		((poffs+=$psz))
+	done
+
+	echo "$(cat _targets/armv7m7-imxrt106x.plo)" >> $PREFIX_BUILD/plo/script.plo
+	echo "PLOEND" >> $PREFIX_BUILD/plo/script.plo
+	echo "00" >> $PREFIX_BUILD/plo/script.plo
```
Comment (agkaminski) [inline _targets/build.project.armv7m7-imxrt106x:69]: Replace with '\0'

**Review by agkaminski — CHANGES_REQUESTED** [review]

File: _targets/build.project.armv7m7-imxrt106x (hunk 1 above)
Comment (HBuczynski) [reply _targets/build.project.armv7m7-imxrt106x]: Done

File: _targets/build.project.armv7m7-imxrt106x [hunk 2]
```diff
@@ -32,9 +33,43 @@ export PORTS_JANSSON=y
...
 #
+
+
+b_mkscript() {
+	ksz=$((($(stat -c "%s" ${PREFIX_PROG_STRIPPED}phoenix-${TARGET}.elf) + $SIZE_PAGE - 1)&$PAGE_MASK))
+	poffs=$(($KERNEL_OFFS + $ksz))
+
+	for prog in "${PROGS[@]}"; do
+		psz=$((($(stat -c "%s" ${PREFIX_PROG_STRIPPED}$prog) + $SIZE_PAGE - 1)&$PAGE_MASK))
+		echo @$prog\($(printf "%x" $poffs):$(printf "%x" $psz)\) >> $PREFIX_BUILD/plo/script.plo
```
Comment (kemonats) [inline _targets/build.project.armv7m7-imxrt106x:63]: simply one printf
```bash
printf "@%s(%x:%x)\n" $prog $poffs $psz >> $PREFIX_BUILD/plo/script.plo
```

File: _targets/build.project.armv7m7-imxrt106x [hunk 3]
```diff
@@ -32,9 +33,43 @@ export PORTS_JANSSON=y
...
+		((poffs+=$psz))
+	done
+
+	echo "$(cat _targets/armv7m7-imxrt106x.plo)" >> $PREFIX_BUILD/plo/script.plo
```
Comment (kemonats) [inline _targets/build.project.armv7m7-imxrt106x:67]: only cat is needed
```bash
cat _targets/armv7m7-imxrt106x.plo >> $PREFIX_BUILD/plo/script.plo
```

File: _targets/build.project.armv7m7-imxrt106x (hunk 2 above)
Comment (HBuczynski) [reply _targets/build.project.armv7m7-imxrt106x]: Done

File: _targets/build.project.armv7m7-imxrt106x (hunk 3 above)
Comment (HBuczynski) [reply _targets/build.project.armv7m7-imxrt106x]: Done

File: _targets/build.project.armv7m7-imxrt106x [hunk 4]
```diff
@@ -49,43 +84,46 @@ b_build_target() {
...
+	PROGS=("dummyfs" "imxrt-multi" "psh" "imxrt-flash")
 
 	# FIXME: build plo out of tree and without clean
 	b_log "Building plo"
-	(cd plo; make clean all)
+	(cd plo; make $MAKEFLAGS clean all)
+
+	b_mkscript
+
+	b_log "Relink plo to include newly script"
```
Comment (agkaminski) [inline _targets/build.project.armv7m7-imxrt106x:96]: *new script

**Pull request merged**